lms/services/html_service.py

```diff
@@ -1,16 +1,23 @@
+import re
 from html.parser import HTMLParser
 
 
-def strip_html_tags(html: str) -> str:
-    """Get plain text from a string which may contain HTML tags."""
+class WhiteSpaceHTMLParser(HTMLParser):  # pylint:disable=abstract-method
+    def __init__(self):
+        super().__init__()
+        self._chunks = []
+
+    def handle_data(self, data):
+        self._chunks.append(data)
+
+    def get_text(self) -> str:
+        joined = "".join(self._chunks).strip()
+        # Remove any superfluous white space added after new lines
+        return re.sub(r"\n\s", "\n", joined)
 
-    # Extract text nodes using HTMLParser. We rely on it being tolerant of
-    # invalid markup.
-    chunks = []
-    parser = HTMLParser()
-    parser.handle_data = chunks.append
+
+def strip_html_tags(html: str) -> str:
+    parser = WhiteSpaceHTMLParser()
     parser.feed(html)
     parser.close()
-
-    # Strip leading/trailing whitespace and duplicate spaces
-    return " ".join("".join(chunks).split())
```

Comment:
This was very specific to the use case in jstor, where it used in titles. This removes newlines.

Move it back to the jstor service.